src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -42,17 +39,21 @@ internal sealed class DependencyUpdateErrorProcessor : IServiceImplementation
         // github url -> https://github.com/maestro-auth-test/maestro-test2  repo ->maestro-test2 owner -> maestro-auth-test2 
         private static readonly Regex RepositoryUriPattern = new Regex(@"^/(?<owner>[^/]+)/(?<repo>[^/]+)/?$");
 
+        public Func<string , GitHubClient> AuthenticateGitHubClient { get; }
+
         public DependencyUpdateErrorProcessor(
             IReliableStateManager stateManager,
             ILogger<DependencyUpdateErrorProcessor> logger,
             BuildAssetRegistryContext context,
-            IOptions<DependencyUpdateErrorProcessorOptions> options
+            IOptions<DependencyUpdateErrorProcessorOptions> options,
+            Func<string, GitHubClient> authenticateGithubClient
```

Comment:
Interesting way to pass the authentication haha. I guess it saves you the trouble of creating a mock class and injecting that instead of the real one during the tests?

Reply:
Will no longer user this :D Will be just passing only the client :)